src/main/java/seedu/loyaltylift/logic/parser/AddPointsCommandParser.java

```diff
@@ -0,0 +1,46 @@
+package seedu.loyaltylift.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.loyaltylift.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.loyaltylift.logic.parser.CliSyntax.PREFIX_POINTS;
+
+import java.util.stream.Stream;
+
+import seedu.loyaltylift.commons.core.index.Index;
+import seedu.loyaltylift.logic.commands.AddPointsCommand;
+import seedu.loyaltylift.logic.parser.exceptions.ParseException;
+import seedu.loyaltylift.model.customer.Points;
+
+/**
+ * Parses input arguments and creates a new AddPointsCommand object
+ */
+public class AddPointsCommandParser implements Parser<AddPointsCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddPointsCommand
+     * and returns an AddPointsCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddPointsCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_POINTS);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_POINTS)
+                || argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddPointsCommand.MESSAGE_USAGE));
+        }
+        Index index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        Points.AddPoints addPoints = ParserUtil.parseAddPoints((argMultimap.getValue(PREFIX_POINTS).get()));
```

Comment:
A simpler solution compared to having an `AddPoints` inner class might be to just return the integer directly in `parseAddPoints` without checking for validity (i.e. allow any integer)

Technically the validity depends on the current point anyway and that is already being validated in `AddPointsCommand::createEditedCustomer` when editing the existing point

Reply:
hmm that is true, I thought there were some benefits to using an `AddPoints` inner class, like encapsulation, and it could also be easily extensible to do `pt/*1000`  because of the enum Modifier. This could signify some other form of modifier to be applied on points.

But I realise in the end I did not even parse the character before the Integer and merely made use of `Integer.valueOf`, so yeah I'll change it to what you suggest, and anyway I don't think we will be adding such functionalities on points, aside from addition and subtraction.

In the end, I still kept a validity check in `parseAddPoints`. I feel like if I allow any integer, some kind of obscure bugs could be found due to  integer overflow or underflow.

Reply:
Ok, sounds good!